website/blog/2024-10-28-kustomize-vs-helm/index.mdx

```diff
@@ -0,0 +1,179 @@
+
+-   Dependency management can be difficult and error-prone.   
+-   Updating CRDs (Custom Resource Definitions) is not supported.   
+-   There's no server-side component, making it hard to track issues like health degradation or configuration drift.   
+-   Native support for reusing configurations with environment-specific changes is lacking.
+    
+
+### Helm chart structure
+
+A typical Helm chart project structure looks like this:
+
+```
+mychart/
+  Chart.yaml            
+  values.yaml         
+  charts/            
+  templates/         
+    deployment.yaml
+    service.yaml
+```
+
+-   `Chart.yaml`: Contains metadata about the chart (name, version, etc.).   
+-   `values.yaml`: Stores the default configuration values, which can be customized and injected into the templates dynamically.   
+-   `templates/`: Holds the Kubernetes manifests for each resource (e.g., Deployment, Service, etc.).
+    
+
+There’s much more to Helm than what’s covered here. As mentioned earlier, a previously published [in-depth guide](/blog/what-is-helm-in-kubernetes) explores Helm in greater detail. In the upcoming sections, we’ll focus on Helm specifically in comparison to Kustomize.
+
+## What is Kustomize?
+
+Kustomize is a Kubernetes-native tool developed after Helm, addressing the need to manage Kubernetes configurations declaratively without templating. Many Kubernetes users sought a way to manipulate plain YAML files with [kubectl](/guides/kubectl/) commands, whether for handling multiple files, creating or updating `ConfigMaps` and `Secrets`, or rolling out changes across environments. Kustomize emerged as a solution for configuration management, using base manifests to create environment-specific variants.
+
+Unlike Helm, which relies on templates, Kustomize uses overlays to apply different configurations to a base set of resources. This allows you to maintain base YAML files and layer environment-specific changes, such as for `development`, `staging`, or `production`.
+
+> We recently published an [in-depth Kustomize tutorial](/blog/patching-with-kustomize/) that explores the various types of patches, transformers, and generators, along with best practices for using the tool effectively. Be sure to check it out for a more detailed guide.
+
+### How to use Kustomize?
+
+Kustomize allows you to manage and customize Kubernetes configurations by layering modifications without altering the original files. A typical Kustomize setup involves a directory structure consisting of a base and overlays.
+
+![How to use Kustomize](https://github.com/user-attachments/assets/1844cdcb-507d-43e6-b5f1-8daaef4e9dd0)
+
+### Kustomise project structure
+
```

Comment:
I think it should be mentioned briefly here, that there isn't a definitive project structure for kustomize. This is just a structure recommended in the Kustomize documentation because it has proven to be quite flexible.